Use briefdescription in autosummary table if it is available

0phoff opened the pull request:
Hey,

Thanks for your awesome work!

I made a small modification to your code to allow the use of the [briefdescription](https://www.doxygen.nl/manual/commands.html#cmdbrief) in the autosummary tables.
If your class/method has a briefdescription it will be used in the autosummary, otherwise it defaults to your old behaviour.

Since you implemented `get_doc` separately in your `DoxygenClassDocumenter` and `DoxygenMethodDocumenter` classes, I did the same for `get_brief`, but maybe it would be nice to reformat that and put both methods straight in your `DoxygenDocumenter` class ? That way there is less code duplication (and you can still overwrite the function later down the line).
Let me know if you agree with this and I will push a new commit with these changes!

Kind Regards,

rmcgibbo commented:
Sounds good to me. I'm not using this project anymore, but maybe @peastman is, so pinging him.

peastman commented:
Yes, I'm still using this.  I'll trust your judgement on whether this is a good change.

0phoff commented:
I also noticed that the `:toctree:`  option of the autosummary class was broken, so I fixed it.

The problem was that the path to the RST files contained `::` separators, while the real generate rst files have '.' separators.
I thus copied the Autosummary.run() method and modified the `real_name` strings to have '.' separators.

Here are the results before and after fixing the issue.
As you can see, the toctree gets correctly build and the Crop class becomes a child page of `docs >> transform`

__Broken__:  
![toctree-broken](https://user-images.githubusercontent.com/11853089/100514152-8c036e80-3172-11eb-8e10-3f4045d0636c.png)

__Fixed__:  
![toctree-fixed](https://user-images.githubusercontent.com/11853089/100514156-8f96f580-3172-11eb-80d4-f7819e85741a.png)
